include/sycl_ext_complex.hpp

```diff
@@ -252,7 +252,7 @@ template<class T> complex<T> tanh (const complex<T>&);
 #endif
 #endif
 
-#if defined(__FAST_MATH__) || defined(_GLIBCXX_FAST_MATH) || defined(_M_FP_FAST)
```

Comment:
why removing `_GLIBCXX_FAST_MATH` ?

Reply:
Because checking for `_GLIBCXX_FAST_MATH` will always expand to true since `_GLIBCXX_FAST_MATH` is always defined as either 0 or 1.
Secondly, because `_GLIBCXX_FAST_MATH` is set to 1, if `__FAST_MATH__` is defined (see https://gcc.gnu.org/onlinedocs/libstdc++/libstdc++-api-4.5/a00766_source.html line 343), which means, no need to check for `_GLIBCXX_FAST_MATH`

Lastly, please pay little attention to the WIP as this PR is still a draft; Even though I fixed the tests locally, the CI is always failing, not because of fast_math or others, but due to an error within the device JIT.

Will continue to look into that, tomorrow

See: 
```
 2:   The program was built for 1 devices
2:   Build program log for 'Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz':
2:   Compilation started
2:   Compilation done
2:   Linking started
2:   Linking done
2:   Device build started
2:   Options used by backend compiler: -g
2:   Failed to build device program
2:   CompilerException Failed to lookup symbol
2:   _ZTSZL31CATCH2_INTERNAL_TEMPLATE_TEST_0IfEvvEUlvE1_
2:   JIT session error: Symbols not found: [ _Z5isinfDh, _Z8isfiniteDh ]
```

Reply:
Oh, nice digging into GLIBC intrinsic! Thanks a lot :) 

>Lastly, please pay little attention to the WIP as this PR is still a draft; Even though I fixed the tests locally, the CI is always failing, not because of fast_math or others, but due to an error within the device compilation.

Oh yes, sorry! Didn't wanted to stress you. Just got the github email and want to reply to them quickly, if not I forgot and never review it.  Will mute this thread :) Sorry about that! 

Thanks again